code/components/extra-natives-five/src/EntityExtraNatives.cpp

```diff
@@ -18,6 +62,72 @@ static const char* GetEntityArchetypeName(int entityHandle)
 	return "";
 }
 
+static HookFunction hookFunction([]()
+{
+	PedIdOffset = *hook::get_pattern<int>("00 00 66 85 C0 78 ? 0F BF C8 0F B7 05 ? ? ? ? 3B C8 7D ? 44 8B C1", -0x2);
+	CReplayPedExtensionPtr = hook::get_address<uintptr_t>(hook::get_pattern<uintptr_t>("4C 03 05 ? ? ? ? EB 03 4D 8B C3"), 3, 7);
+
+	fx::ScriptEngine::RegisterNativeHandler("GET_PED_DECORATIONS", [](fx::ScriptContext& context)
+	{
+		if (PedIdOffset == 0x0 || CReplayPedExtensionPtr == 0x0)
+		{
+			context.SetResult(fx::SerializeObject(std::vector<DecorationResult>()));
+			return;
+		}
+
+		fwEntity* entity = rage::fwScriptGuid::GetBaseFromGuid(context.GetArgument<int>(0));
+		std::vector<DecorationResult> decorationList;
+
+		if (entity && entity->IsOfType<CPed>())
+		{
+			uint16_t bufferIndex = *(uint16_t*)((uintptr_t)entity + PedIdOffset);
+			ptrCReplayPedExtension* decorationSystem = (ptrCReplayPedExtension*)CReplayPedExtensionPtr;
+			if (decorationSystem && decorationSystem->ptr)
+			{
+				PedEntry* decorationPedEntry = &decorationSystem->ptr->entries[bufferIndex];
+				if (decorationPedEntry)
+				{
+					decorationList.resize(decorationPedEntry->overlay_count);

```

Comment:
Perhaps prefer `.reserve()` and `.push_back` in the loop (or even `.emplace_back` without the `{}` initializers).